File: app/src/main/java/org/jellyfin/androidtv/ui/playback/VlcEventHandler.java
package org.jellyfin.androidtv.ui.playback;

import org.videolan.libvlc.MediaPlayer.Event;

public class VlcEventHandler implements org.videolan.libvlc.MediaPlayer.EventListener {

    private PlaybackListener onCompletionListener;
    private PlaybackListener onErrorListener;
    private PlaybackListener onPreparedListener;
    private PlaybackListener onProgressListener;

    public void setOnCompletionListener(PlaybackListener listener) {
        onCompletionListener = listener;
    }

    public void setOnErrorListener(PlaybackListener listener) {
        onErrorListener = listener;
    }

    public void setOnPreparedListener(PlaybackListener listener) {
        onPreparedListener = listener;
    }

    public void setOnProgressListener(PlaybackListener listener) {
        onProgressListener = listener;
    }

    @Override
    public void onEvent(Event event) {
        switch (event.type) {
            case Event.EndReached:
                if (onCompletionListener != null) onCompletionListener.onEvent();
                break;
            case Event.Playing:
                if (onPreparedListener != null) onPreparedListener.onEvent();
                break;
            case Event.PositionChanged:
                if (onProgressListener != null) onProgressListener.onEvent();
                break;
            case Event.EncounteredError:
                if (onErrorListener != null) onErrorListener.onEvent();
            case Event.Paused:
            case Event.Stopped:
            default:
                break;

        }
    }
}

File: app/src/main/java/org/jellyfin/androidtv/ui/playback/VlcEventHandler.kt
package org.jellyfin.androidtv.ui.playback

import org.videolan.libvlc.MediaPlayer

class VlcEventHandler : MediaPlayer.EventListener {
	var onCompletionListener: PlaybackListener? = null
	var onErrorListener: PlaybackListener? = null
	var onPreparedListener: PlaybackListener? = null
	var onProgressListener: PlaybackListener? = null

	override fun onEvent(event: MediaPlayer.Event) {
		when (event.type) {
			MediaPlayer.Event.EndReached -> onCompletionListener?.onEvent()
			MediaPlayer.Event.Playing -> onPreparedListener?.onEvent()
			MediaPlayer.Event.PositionChanged -> onProgressListener?.onEvent()
			MediaPlayer.Event.EncounteredError -> onErrorListener?.onEvent()
			else -> Unit
		}
	}
}
